tests/testthat/test-hospital-model.R

```diff
@@ -0,0 +1,265 @@
+    mod <- sircovid_model$odin_model(user = pars_model)
+    t_max <- 150
+    t <- seq(from = 1, to = t_max)
+    tmp <- mod$run(t)
+    results <- mod$transform_variables(tmp)
+     #should be TRUE
+    expect_true(all(results$S[t_max,] == pars_model$S0))
+
+  }
+)
+
+test_that("everyone is infected when beta is Inf", {
+  
+    sircovid_model <- hospital_model()
+    pars_model <- generate_parameters(sircovid_model = sircovid_model, beta = Inf)
+    mod <- sircovid_model$odin_model(user = pars_model)
+    t_max <- 150
+    t <- seq(from = 1, to = t_max)
+    tmp <- mod$run(t)
+    results <- mod$transform_variables(tmp)
+     #should be TRUE
+    expect_true(all(results$S[2:t_max,] == 0))
+ }
+
+)
+
+
+test_that("No one is infected if I and E are 0 at t = 0", {
+  
+    sircovid_model <- hospital_model()
+    pars_model <- generate_parameters(sircovid_model = sircovid_model, beta = 0.042)
+    pars_model$E0[,,]<- 0
+    pars_model$I0_asympt[,,]<- 0
+    pars_model$I0_mild[,,]<- 0
+    pars_model$I0_ILI[,,] <- 0
+    pars_model$I0_hosp_R[,,] <- 0
+    pars_model$I0_hosp_D[,,] <- 0
+    pars_model$I0_ICU_R[,,] <- 0
+    pars_model$I0_ICU_D[,,] <- 0
+    pars_model$I0_triage[,,] <- 0
+
+    mod <- sircovid_model$odin_model(user = pars_model)
+    t_max <- 150
+    t <- seq(from = 1, to = t_max)
+    tmp <- mod$run(t)
+    results <- mod$transform_variables(tmp)
+    expect_true(all(results$S[t_max,] == pars_model$S0))
+
+ }
+)
+
+test_that("No one is hospitalised if p_sympt_ILI is 0", {
+    
+    sircovid_model <- hospital_model()
+    pars_model <- generate_parameters(sircovid_model = sircovid_model, beta = 0.042)
+    pars_model$p_sympt_ILI[]<-0
+    mod <- sircovid_model$odin_model(user = pars_model)
+    t_max <- 150
+    t <- seq(from = 1, to = t_max)
+    check_cases <- FALSE
+    max_iter <- 10
+    iter <- 0
+    while (!check_cases && iter<= max_iter){
+      iter <- iter + 1
+      tmp <- mod$run(t)
+      results <- mod$transform_variables(tmp)
+      if (any(results$E[,,,] > 0)){
+        check_cases <- TRUE
+      }
+    }
+    #check there were people infected (if yes, should be TRUE)
+    expect_true(all(results$I_ILI[,,,] == 0))
+    expect_true(all(results$I_hosp_R[,,,] == 0))
+    expect_true(all(results$I_hosp_D[,,,] == 0))
+    expect_true(all(results$I_ICU_R[,,,] == 0))
+    expect_true(all(results$I_ICU_D[,,,] == 0))
+    expect_true(all(results$I_triage[,,,] == 0))
+    expect_true(all(results$R_stepdown[,,,] == 0))
+    expect_true(all(results$D[,]==0))
+ }
+)
+
+
+test_that("No one is hospitalised if p_recov_ILI is 0", {
+  #p_recov_ILI=1, no-one hospitalised
+  sircovid_model <- hospital_model()
+  pars_model <- generate_parameters(sircovid_model = sircovid_model, beta = 0.126)
+  pars_model$p_recov_ILI[]<-1
+  mod <- sircovid_model$odin_model(user = pars_model)
+  t_max <- 150
+  t <- seq(from = 1, to = t_max)
+  check_cases <- FALSE
+  max_iter <- 10
+  iter <- 0
+  while (!check_cases && iter<= max_iter){
```

Comment:
Can you add a comment to note why the loops are being used.
Also I think putting an `expect_true(any(results$I_ILI[,,,] > 0))` after the loop would make sense, to make sure if it exits after max_iter the correct failure is diagnosed

Reply:
Have added comments about why the loops are being used. Added expect_true statements like you suggested after every while loop. Also switched p_death_hosp to p_death_hosp_D in line with the changes in 0.2.18.